src/Workleap.DomainEventPropagation.Subscription.PullDelivery/EventPropagationSubscriptionOptions.cs

```diff
@@ -24,4 +24,14 @@ public class EventPropagationSubscriptionOptions
     public int MaxRetries { get; set; } = 3;
 
     public IReadOnlyCollection<TimeSpan>? RetryDelays { get; set; }
+
+    /// <summary>
+    /// Maximum number of events to pull from Event Grid per poll.
+    /// </summary>
+    public int MaxPullBatchSize { get; set; } = 100;
+
+    /// <summary>
+    /// Interval to wait between polling for new events.
+    /// </summary>
+    public TimeSpan PullInterval { get; set; } = TimeSpan.Zero;
```

Comment:
I think it would be better to use a rate limiting solution: https://devblogs.microsoft.com/dotnet/announcing-rate-limiting-for-dotnet/?WT.mc_id=DT-MVP-5003978

This would support any scenario needed by the client.